### PR TITLE
[action] plugin_scores: Add support for rendering multiline description of plugin actions on docs

### LINKED ISSUE #15201: plugin_scores: Fastlane docs parse only the last line of multiline action Description
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

On docs.fastlane.tools, action description is only the last line when multi-line Ruby string is used:

https://github.com/fastlane/docs/blob/93169702a4f3c090824c08747f3b7050f3094ee1/plugin_scores_cache.yml#L4560

vs

https://github.com/dotdoom/fastlane-plugin-flutter/blob/23d517049d11da95563d35f76bfd44a64eb54eeb/lib/fastlane/plugin/flutter/actions/flutter_generate_action.rb#L36-L39

### PR DESCRIPTION
**Pull request by mollyIV**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context

> On docs.fastlane.tools, action description is only the last line when multi-line Ruby string is used:
> 
> https://github.com/fastlane/docs/blob/93169702a4f3c090824c08747f3b7050f3094ee1/plugin_scores_cache.yml#L4560
> 
> vs
>
>https://github.com/dotdoom/fastlane-plugin-flutter/blob/23d517049d11da95563d35f76bfd44a64eb54eeb/lib/fastlane/plugin/flutter/actions/flutter_generate_action.rb#L36-L39

closes https://github.com/fastlane/fastlane/issues/15201

### Description

When a method called `description` is detected, we basically concat description lines till the end of the block. 

I generated the docs locally and multiline descriptions look good 👍 

**Important note** ⚠️ 

The following description won't be rendered correctly on the docs site due to breaking markdown `${...}`

https://github.com/dotdoom/fastlane-plugin-flutter/blob/23d517049d11da95563d35f76bfd44a64eb54eeb/lib/fastlane/plugin/flutter/actions/flutter_generate_action.rb#L35-L40

I modified the description locally:

```
<details>
<summary class="plugin-actions">Provided actions</summary>

 Name | Category | Description
------|----------|------------
  **flutter_bootstrap** | - | Flutter SDK installation, upgrade and application bootstrap
  **flutter** | - | Run "flutter" binary with the specified arguments
  **flutter_build** | building | Run "flutter build" to build a Flutter application
  **flutter_generate** | - | According to package:intl, take `$strings_file` and generate `${strings_file.dirname}/arb/intl_messages.arb`, then take all files matching `${strings_file.dirname}/intl_*.arb`, fix them and generate .dart files from them

</details>
```

 and got:

Screen Shot 2019-08-21 at 09 38 05

### Testing steps

To test these changes, please point a `fastlane` version in your `Gemfile` to my branch and follow the instructions from [docs repository](https://github.com/fastlane/docs) 🙇

### CODE REVIEW

**Comment by janpio** [comment]
I don't understand the "Important note". Do plugin authors have to change anything? Or do we change things to fix this?

**Comment by mollyIV** [comment]
> I don't understand the "Important note". Do plugin authors have to change anything? 

Plugin author will need to make sure that he/she does not break markdown syntax, i.e. 

```
${strings_file.dirname}
``` 

It should've been 
```
`${strings_file.dirname}/`
```

> Or do we change things to fix this?
Nothing to be changed on our side, this PR is all we need to introduce a multiline description support 👍

**Comment by janpio** [comment]
> Plugin author will need to make sure that he/she does not break markdown syntax, i.e.
> 
> ```
> ${strings_file.dirname}
> ```
> 
> It should've been
> 
> ```
> `${strings_file.dirname}/`
> ```

I might be missing something, how is the first one not valid markdown?

**Comment by janpio** [comment]
Also notify @dotdoom about this PR.

**Comment by mollyIV** [comment]
> I might be missing something, how is the first one not valid markdown?

Sorry @janpio for misleading information and not providing full context. 

This is rendering correctly and markdown is absolutely valid 👇 

 Name | Category | Description
------|----------|------------
  **flutter_bootstrap** | - | Flutter SDK installation, upgrade and application bootstrap
  **flutter** | - | Run "flutter" binary with the specified arguments
  **flutter_build** | building | Run "flutter build" to build a Flutter application
  **flutter_generate** | - | According to package:intl, take $strings_file and generate ${strings_file.dirname}/arb/intl_messages.arb, then take all files matching ${strings_file.dirname}/intl_*.arb, fix them and generate .dart files from them

However, the very same table is rendering incorrectly when I run it locally in a browser via `mkdocs serve`:

image

```HTML
<blockquote>

<p>Flutter actions plugin for Fastlane</p>
</blockquote>
<p><details>
<summary class="plugin-actions">Provided actions</summary></p>
<table>
<thead>
<tr>
<th>Name</th>
<th>Category</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><strong>flutter_bootstrap</strong></td>
<td>-</td>
<td>Flutter SDK installation, upgrade and application bootstrap</td>
</tr>
<tr>
<td><strong>flutter</strong></td>
<td>-</td>
<td>Run "flutter" binary with the specified arguments</td>
</tr>
<tr>
<td><strong>flutter_build</strong></td>
<td>building</td>
<td>Run "flutter build" to build a Flutter application</td>
</tr>
<tr>
<td><strong>flutter_generate</strong></td>
<td>-</td>
<td strings_file.dirname="strings_file.dirname">According to package:intl, take $strings_file and generate $</td>
</tr>
</tbody>
</table>
<p></details></p>
<p><details>
<summary class="score-details">Score details</summary></p>
```

Do we generate the docs in the same way? 
It looks that it may be related to `mkdocs`.

**Comment by janpio** [comment]
Yes, that might indeed be a mkdocs problem then.

So the generated markdown from the action is correct, and just the rendered html from mkdocs is broken - then this would be a fastlane/docs issue we could look into in the future.

File: fastlane/helper/plugin_scores_helper.rb [hunk 1]
```diff
@@ -325,7 +326,13 @@ def parse_file(file, debug_state: false)
               end
             when :in_method
               if (string_statement = string_statement_from_line(line))
-                last_string_statement = string_statement
+                # Multiline support for `description` method
+                if last_method_name == 'description'
+                  # rubocop:disable BlockNesting
+                  last_string_statement.concat(string_statement) unless last_string_statement.nil?
```
Comment (joshdholtz) [inline fastlane/helper/plugin_scores_helper.rb]: Won't `last_string_statement` always be `nil` here? 🤔 It gets `nil`-ed on line 309 and 344.

**Review by joshdholtz — COMMENTED** [review]
This looks great! Juts one small question 🙃

File: fastlane/helper/plugin_scores_helper.rb (hunk 1 above)
Comment (mollyIV) [reply fastlane/helper/plugin_scores_helper.rb]: Good question 👍 

It gets `nil`-ed on the end of the line. The regex to detect the end of the line looks like this:

```
def is_end?(line)
    !!line.match(/^\s*end\s*/)
end
```

So if we have a multi-line description, the end of the line is not detected and `last_string_statement` is not nil then 😉

**Review by joshdholtz — APPROVED** [review]
Looks good to me! 💪 Thanks for adding this!

**Pull request merged**